github/repos_codeowners.go

```diff
@@ -0,0 +1,41 @@
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// CodeownersErrors represents a list of syntax errors detected in the CODEOWNERS file.
+type CodeownersErrors struct {
+	Errors []*CodeownersError `json:"errors"`
+}
+
+// CodeownersError represents a syntax error detected in the CODEOWNERS file.
+type CodeownersError struct {
```

Comment:
Looking at the official docs for this data, it appears that all the fields will be returned except the `suggestion` field, so I'm thinking that we might want to break from the surrounding practice in this repo and make the rest of these fields values instead of pointers.

Reply:
Changed that. Do you think
```go
// CodeownersErrors represents a list of syntax errors detected in the CODEOWNERS file.
type CodeownersErrors struct {
	Errors []*CodeownersError `json:"errors"`
}
```
should also be a slice of values instead of a slice of pointers?

Reply:
No, I think that is fine because when we have slices of structs, we typically use pointers to reduce the chance of errors when iterating over the items... so let's leave this one.  Thanks!